### PR TITLE
docs: update ! to better reflect how it is  used in the wild

### PR DESCRIPTION
**Pull request by bcoe**
Now that I've had some experience using `!` in the wild, my opinion of how we should define the feature has changed a bit.

There are two incidents on my mind:

1. myself and my peers released a breaking change as a minor because we accidentally didn't rewrite the merge message (dropping the BREAKING CHANGE that we had in the description); this is currently fairly easy to do with GitHub's UI.
1. motivated by the above incident, we started using `!` in our commit messages -- however, history repeated itself and we merged a commit with `!` in the header, but no BREAKING CHANGE in body.

In the process of these two incidents, I came around to @DominicKramer and @ofrobots' argument that having an indicator of a breaking change in the header of the commit message itself should be enough to represent a breaking change -- it would have saved us from both accidents.

Long story short:

I'd like to advocate that we fallback to the description of the commit message as a description of a breaking change, if no BREAKING CHANGE is provided in the body or footer; this makes it so that `!` can be used as an alternative to BRAKING CHANGE.

I'd also like to advocate that we consider moving to version 1.0.0 of the specification (CC: @damianopetrungaro). I'm feeling like we've had a great chance to play with the specification in a variety of production environments and it has reached a healthy point of maturity.

CC: @JustinBeckwith, @zeke, @apetro,

### CODE REVIEW

File: content/next/index.md [hunk 1]
```diff
@@ -91,12 +96,14 @@ The description is a short summary of the code changes, e.g., _fix: array parsin
 1. A footer of one or more lines MAY be provided one blank line after the body. The footer MUST contain meta-information
 about the commit, e.g., related pull-requests, reviewers, breaking changes, with one piece of meta-information
 per-line.
-1. Breaking changes MUST be indicated at the very beginning of the body section, or at the beginning of a line in the footer section. A breaking change MUST consist of the uppercase text BREAKING CHANGE, followed by a colon and a space.
-1. A description MUST be provided after the `BREAKING CHANGE: `, describing what has changed about the API, e.g., _BREAKING CHANGE: environment variables now take precedence over config files._
+1. Breaking changes MUST be indicated in the type/scope prefix of a commit, or at the beginning of its optional body or footer section.
+1. If included in the body or footer, a breaking change MUST consist of the uppercase text BREAKING CHANGE, followed by a colon, space, and description, e.g.,
```
Comment (bcoe) [inline content/next/index.md]: maybe we should call out the fact that a reason for using `BREAKING CHANGE` would be that you wish to provide a longer description of the breaking change.

**Comment by apetro** [comment]
@bcoe I think what you're saying is you've experienced how important it is that the breaking change be indicated in the header of the commit message.

So I'd go further, please.

Make the `!` in the header REQUIREd of breaking changes.  Make the BREAKING CHANGE in the body of the commit message optional.

This will be less complex than what this Pull Request presently proposes.

Firstly, making the `!` REQUIREd will mean that there's one reliable indicator of breaking changes to look for. As this Pull Request stands, it would make either of two indicators valid, which means people and tooling looking for known breaking changes would have to look at both the header and the body of commit messages to find all the known breaking changes in a range.

Secondly, in general the header of the commit message is required and the body and footer of commit messages are optional, in Conventional Commits and generally in git practice. This Pull Request would leave it that there's this one edge case where the body is required: when it's a breaking change and that breakingness wasn't indicated in the header. That optional-except-when-it's-required-depending-on-what-choice-you-made-about-this-other-optional-except-when-it's-required-thing is complicated. It's simpler to say that the `!` in the header is required, and the body is optional, full stop.

**Comment by apetro** [comment]
For whatever it's worth, I'm 👎 on releasing 1.0.0 *at this time*. This PR demonstrates that the indication of breaking changes is still being figured out. Indicating breaking changes is the very most important thing a commit message can communicate. It's the very most important thing to get right in Conventional Commits. So, the path forward is, iterate the specification about breaking changes. Ship another release candidate. Get experience with that release candidate in the wild.

Conventional Commits should ship a 1.0.0 that is literally the same as a previously shipped release candidate that's demonstrated successful in the wild. No sneaking last minute changes into the spec. :)

**Comment by damianopetrungaro** [comment]
@apetro  we already discussed.

We got your point and we agree, but we can't make it required until the tools and the community will start using it.

**Comment by zeke** [comment]
I love this idea! I think @apetro's suggestion is on-point too, and removes ambiguity:

> Make the ! in the header REQUIREd of breaking changes. Make the BREAKING CHANGE in the body of the commit message optional.

👍

**Comment by bcoe** [comment]
@zeke, @apetro, I agree that it simplifies the specification to make `!` required, and I have been moving towards using `!` in my PRs.

My concern echos @damianopetrungaro's point; making `!` required would mean that we're no longer in agreement with `angular` conventions, which the majority of the tooling in the community is written to support -- maybe the compromise is to land this edit to the specification, but not move to `1.x.x` of the specification for a while? see how the community adopts the `!`.

**Comment by zeke** [comment]
> maybe the compromise is to land this edit to the specification, but not move to 1.x.x of the specification for a while?

okay with me. baby steps 👍

**Comment by damianopetrungaro** [comment]
We are discussing the same things we already discussed time ago :)
Let's just see the userland usage,after we understand it we can bump it up to the 1.0.0 but there's no rush atm.
